### PR TITLE
Kerberos asrep roasting improvements

### PR DESCRIPTION
**Pull request by adfoster-r7**
Makes improvements to the asrep roasting module

Bug fix:

```
[+] [2024.07.22-23:57:58] Workspace:test Beginning step 1/1 Scanning 192.168.123.133-192.168.123.133 - Progress: 0%
[*] [2024.07.22-23:57:58] Using domain: DEMO.LOCAL - 192.168.123.133:88   ...
[*] [2024.07.22-23:57:58] 192.168.123.133 - User: "user1" user not found
[+] [2024.07.22-23:57:58] 192.168.123.133 - User: "vagrant" is present
[!] [2024.07.22-23:57:58] 192.168.123.133:88    - LOGIN FAILED: {:private_data=>nil, :private_type=>:password, :username=>"test_user", :realm_key=>"Active Directory Domain", :realm_value=>"DEMO.LOCAL"} - Unhandled error - scan may not produce correct results: undefined method `find' for nil:NilClass

              etype_entries = pa_data.find {|entry| entry.type == Rex::Proto::Kerberos::Model::PreAuthType::PA_ETYPE_INFO2}
                                     ^^^^^ - 
[+] [2024.07.22-23:57:58] 192.168.123.133 - User: "administrator" is present
[*] [2024.07.22-23:57:58] Complete (0 sessions opened) auxiliary/gather/kerberos_enumusers
[+] [2024.07.22-23:57:58] Workspace:test Task Completed - Progress: 100%
```

## Verification

- Create an ASREP roastable account in your AD environment

image

- Verify `gather/asrep` and `gather/kerberos_enumusers` can detect this user account

### CODE REVIEW

**Review by dwelch-r7 — APPROVED** [review]

File: modules/auxiliary/gather/asrep.rb [hunk 1]
```diff
@@ -77,26 +77,36 @@ def run
   def run_brute
     result_count = 0
     user_file = datastore['USER_FILE']
-    if user_file.nil?
-      fail_with(Msf::Module::Failure::BadConfig, 'User file must be specified when brute forcing')
+    username = datastore['USERNAME']
+    if user_file.blank? && username.blank?
+      fail_with(Msf::Module::Failure::BadConfig, 'User file or username must be specified when brute forcing')
+    end
+    if username.present?
```
Comment (adfoster-r7) [inline modules/auxiliary/gather/asrep.rb:84]: Paper trail: I was aligning this module to more closely align with how login scanners work, where you can provide a user file, or a username.

This might be confusing for the auth scenario to LDAP - we can revert this change if so :+1:

**Comment by adfoster-r7** [comment]
## Release Notes

Improves error handling and progress tracking in the `auxiliary/gather/kerberos_enumusers` and `gather/asrep` modules
